modules/ref_fb_module/src/statistics_fb_impl.cpp

```diff
@@ -310,6 +351,63 @@ void StatisticsFbImpl::processDataPacket(const DataPacketPtr& packet)
     domainSignal.sendPacket(outDomainPacket);
 }
 
+void StatisticsFbImpl::processDataPacketInput(const DataPacketPtr& packet)
+{
+    if (!valid)
+        return;
+
+    const auto domainPacket = packet.getDomainPacket();
+    if (!domainPacket.assigned())
+        return;
+
+    if (triggerMode)
+    {
+        const auto packetBuf = static_cast<uint8_t*>(packet.getData());
+        auto domainBuf = static_cast<Int*>(domainPacket.getData());
+
+        size_t i = 0;
+        while (i < packet.getSampleCount())
+        {
+            // Go to where trigger state is true or end of packet
+            while (!triggerHistory.getTriggerStateFromDomainValue(domainBuf[i]) && i < packet.getSampleCount())
+                i++;
```

Comment:
I think there's still a TODO for dropping the triggerHistory, but it should probably be dropped within/after this while loop.